python_modules/libraries/dagster-tableau/dagster_tableau/resources.py

```diff
@@ -436,20 +468,30 @@ def fetch_tableau_workspace_data(
                                 properties=augmented_sheet_data,
                             )
                         )
-
                     for embedded_data_source_data in sheet_data.get(
-                        "parentEmbeddedDatasources", []
+                            "parentEmbeddedDatasources", []
                     ):
-                        for published_data_source_data in embedded_data_source_data.get(
-                            "parentPublishedDatasources", []
-                        ):
-                            data_source_id = published_data_source_data["luid"]
+                        published_data_source_list = embedded_data_source_data.get("parentPublishedDatasources", [])
+                        if len(published_data_source_list) > 0:
+                            for published_data_source_data in published_data_source_list:
+                                data_source_id = published_data_source_data["luid"]
+                                if data_source_id and data_source_id not in data_source_ids:
+                                    data_source_ids.add(data_source_id)
+                                    data_sources.append(
+                                        TableauContentData(
+                                            content_type=TableauContentType.DATA_SOURCE,
+                                            properties=published_data_source_data,
+                                        )
+                                    )
+                        else:
+                            data_source_id = embedded_data_source_data["id"]
                             if data_source_id and data_source_id not in data_source_ids:
                                 data_source_ids.add(data_source_id)
+                                embedded_data_source_data["luid"] = data_source_id
```

Comment:
Could we add a comment here explaining why we use `luid`?

Reply:
Yeah that makes sense! I meant, could we add a small comment in the code? 

```suggestion
                                # your comment here
                                embedded_data_source_data["luid"] = data_source_id
```